### PR TITLE
remove host port declare in chaos-daemon-daemonset.yaml

### PR DESCRIPTION
**Pull request by xlgao-zju**
### What problem does this PR solve?

background ref #1206

### Related changes

- [ ] Need to update `chaos-mesh/website`
- [ ] Need to update `Dashboard UI`
- Need to **cheery-pick to release branches**
  - [ ] release-2.1
  - [ ] release-2.0

### Checklist

Tests

- [ ] Unit test
- [x] E2E test
- [ ] No code
- [ ] Manual test (add steps below)

Side effects

- [ ] Breaking backward compatibility

### Release note 

```text
remove host port declare in chaos-daemon-daemonset
```

### DCO

If you find the DCO check fails, please run commands like below (Depends on the actual situations. For example, if the failed commit isn't the most recent) to fix it:

```shell
git commit --amend --signoff
git push --force
```

### CODE REVIEW

File: helm/chaos-mesh/templates/chaos-daemon-daemonset.yaml [hunk 1]
```diff
@@ -132,7 +132,6 @@ spec:
           ports:
             - name: grpc
               containerPort: {{ .Values.chaosDaemon.grpcPort }}
-              hostPort: {{ .Values.chaosDaemon.grpcPort }}
```
[the remark this answers is not in the document]
Comment (xlgao-zju) [reply helm/chaos-mesh/templates/chaos-daemon-daemonset.yaml:135]: @cwen0 updated

**Review by cwen0 — APPROVED** [review]
LGTM

**Comment by xlgao-zju** [comment]
ping @YangKeao @STRRL @FingerLeader for review

**Review by FingerLeader — APPROVED** [review]
LGTM

**Pull request merged**